common/lib/dependabot/pull_request_creator/github.rb

```diff
@@ -418,23 +423,42 @@ def github_client_for_source
 
       def handle_error(err)
         case err
+        when AnnotationError
+          cause = err.cause
+          raiser = lambda do |type, message|
```

Comment:
The wrapping exception pattern, and this snippet for re-wrapping the delegated exceptions may be 🙃 .

In gopher terms, I was wanting to replace `return nil, err` with `return pr, err`.

Reply:
It took me a few reads to grok this, wondering if it would it be easier to understand if this were a method instead of a lambda?